File: Android_java2/app/src/main/java/com/example/android_java/Intent2.java
package com.example.android_java;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

public class Intent2 extends AppCompatActivity {
    TextView text;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_intent2);

        text=findViewById(R.id.name);
        Intent intent=getIntent();
        String name=intent.getStringExtra("nameData");
        text.setText("My name is "+name);


    }
}

File: Exam_prep/app/src/main/java/com/tecwec/exam_prep/intent2.kt
package com.tecwec.exam_prep

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView

class intent2 : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_intent2)
        var data:TextView=findViewById(R.id.getData)
        var text: String? = intent.getStringExtra("data").toString()
        data.text = text

    }
}